### ISSUE TITLE
ansible-lint is not compatible with community.general tasks

### ISSUE THREAD

**Issue by scrhicks** [ansible/ansible-lint#3842]
##### Summary
I have configured pre-commit and when I'm trying validate my playbook I got an error: `couldn't resolve module/action 'community.general.homebrew_cask'`

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT
Apple M1 Pro (Sonoma 14.0)

```console (paste below)
ansible --version
ansible [core 2.15.5]
  config file = None
  configured module search path = ['/Users/ppogorzelec/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/homebrew/lib/python3.11/site-packages/ansible
  ansible collection location = /Users/ppogorzelec/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/homebrew/bin/ansible
  python version = 3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)] (/opt/homebrew/opt/python@3.11/bin/python3.11)
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version
ansible-lint 6.20.3 using ansible-core:2.15.5 ansible-compat:4.1.10 ruamel-yaml:0.17.22 ruamel-yaml-clib:None

pre-commit --version
pre-commit 3.5.0

ansible-galaxy collection list 
# /opt/homebrew/lib/python3.11/site-packages/ansible_collections
Collection                    Version
----------------------------- -------
...
community.general             7.5.0 
...
```

- ansible installation method: pip
- ansible-lint installation method: pip
- pre-commit installation method: pip

##### STEPS TO REPRODUCE
`sample-playbook.yml`
```yaml
- name: Multipass - Prepare machines and inventory file
  hosts: localhost
  connection: local

  vars_files:
    - vars/multipass.yml

  tasks:
    - name: Multipass - Requirements
      block:
        - name: 'Multipass - Requirements: Ensure multipass is intalled'
          community.general.homebrew_cask:
            name: multipass
            state: present
```

`.pre-commit-config.yaml`
```yaml
repos:
  - repo: https://github.com/ansible-community/ansible-lint.git
    rev: v6.20.3
    hooks:
      - id: ansible-lint
        files: \.(yaml|yml)$
```

```console
pre-commit run -a
```

```console (paste below)
INFO     Identified /Users/ppogorzelec/projects/github/pogosoftware/hashicorp-playground as project root due .git directory.
INFO     Set ANSIBLE_LIBRARY=/Users/ppogorzelec/.cache/ansible-compat/2087ed/modules:/Users/ppogorzelec/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/ppogorzelec/.cache/ansible-compat/2087ed/collections:/Users/ppogorzelec/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/Users/ppogorzelec/.cache/ansible-compat/2087ed/roles:/Users/ppogorzelec/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Excluded: .git
INFO     Excluded: .github
INFO     Set ANSIBLE_LIBRARY=/Users/ppogorzelec/.cache/ansible-compat/2087ed/modules:/Users/ppogorzelec/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/ppogorzelec/.cache/ansible-compat/2087ed/collections:/Users/ppogorzelec/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/Users/ppogorzelec/.cache/ansible-compat/2087ed/roles:/Users/ppogorzelec/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Executing syntax check on playbook ansible/00_multipass_deployment.yml (0.21s)
WARNING  Listing 1 violation(s) that are fatal
syntax-check[specific]: couldn't resolve module/action 'community.general.homebrew_cask'. This often indicates a misspelling, missing collection, or incorrect module path.
ansible/00_multipass_deployment.yml:12:11


                  Rule Violation Summary                   
 count tag                    profile rule associated tags 
     1 syntax-check[specific] min     core, unskippable    

Failed: 1 failure(s), 0 warning(s) on 9 files.
```

##### Desired Behavior
It should pass without errors. Ansible can run this playbook without any warnings or errors.

##### Actual Behavior
ansible-lint can't find a `community.general.homebrew_cask` task

**Comment by ssbarnea** [comment]
@scrhicks please look at #3856 as it should clarify the usage with pre-commit hook. Mainly is not a bug, but a previously undocumented side effect of using ansible bundle. I hope that this sorts your issue.
